### ISSUE TITLE
"Delete unused variables" doesn't work if a method is named like the variable

### ISSUE THREAD

**Issue by ConjuringCoffee** [SAP/abap-cleaner#222]
Hi Jörg-Michael, I found an issue with the rule "Delete unused variables".

Here's a scenario in which it works correctly.

Before:

```abap
REPORT yexample.
CLASS lcl_ DEFINITION CREATE PRIVATE.

  PUBLIC SECTION.
    METHODS test1.

ENDCLASS.


CLASS lcl_ IMPLEMENTATION.
  METHOD test1.
    SELECT
      FROM e071
      FIELDS activity
      INTO TABLE @DATA(test).

    test1( ).
  ENDMETHOD.
ENDCLASS.
```

After:

```abap
REPORT yexample.
CLASS lcl_ DEFINITION CREATE PRIVATE.

  PUBLIC SECTION.
    METHODS test1.
ENDCLASS.


CLASS lcl_ IMPLEMENTATION.
  METHOD test1.
    SELECT
      FROM e071
      FIELDS activity
      " TODO: variable is assigned but never used (ABAP cleaner)
      INTO TABLE @DATA(test).

    test1( ).
  ENDMETHOD.
ENDCLASS.
```

It doesn't work if the method is named like the variable. This code is unchanged:

```abap
REPORT yexample.
CLASS lcl_ DEFINITION CREATE PRIVATE.

  PUBLIC SECTION.
    METHODS test.
ENDCLASS.


CLASS lcl_ IMPLEMENTATION.
  METHOD test.
    SELECT
      FROM e071
      FIELDS activity
      INTO TABLE @DATA(test).

    test( ).
  ENDMETHOD.
ENDCLASS.
```

Thanks! 🙂

**Comment by jmgrassau** [comment]
Hi ConjuringCoffee, 

thanks a lot for perfect documentation of this issue! This bug will be fixed with the next release.

Kind regards,
Jörg-Michael

**Comment by jmgrassau** [comment]
Hi ConjuringCoffee, 

this should now be fixed with version 1.13.0!

Kind regards,
Jörg-Michael
